File: examples/java/pact_validation_with_oas/one_consumer_one_provider/provider/src/main/java/br/com/zup/pact/provider/service/impl/AccountServiceImpl.java
package br.com.zup.pact.provider.service.impl;

import br.com.zup.pact.provider.dto.AccountDetailsDTO;
import br.com.zup.pact.provider.dto.BalanceDTO;
import br.com.zup.pact.provider.repository.AccountRepository;
import br.com.zup.pact.provider.service.AccountService;
import java.util.List;
import java.util.Optional;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class AccountServiceImpl implements AccountService {

    private final AccountRepository accountRepository;

    @Override
    public Optional<AccountDetailsDTO> getAccountDetailsByAccountId(Integer accountId) {
        return accountRepository.findByAccountId(accountId);
    }

    @Override
    public Optional<List<AccountDetailsDTO>> getAll() {
        return accountRepository.getAll();
    }

    @Override
    public Optional<BalanceDTO> getBalanceByAccountId(Integer accountId) {
        return accountRepository.getBalanceByAccountId(accountId);
    }
}

File: example/java/kotlin/provider/src/main/kotlin/br/com/zup/pact/provider/service/AccountServiceImpl.kt
package br.com.zup.pact.provider.service

import br.com.zup.pact.provider.dto.AccountDetailsDTO
import br.com.zup.pact.provider.dto.BalanceDTO
import br.com.zup.pact.provider.repository.AccountRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import java.util.*

@Service
class AccountServiceImpl(@Autowired val accountRepository: AccountRepository) : AccountService {

    override fun getAll(): List<AccountDetailsDTO> {
        return accountRepository.getAll()
    }

    override fun findByAccountId(accountId: Int): Optional<BalanceDTO> {
        return accountRepository.findByAccountId(accountId)
    }
}